app/src/main/kotlin/edu/stanford/bdh/engagehf/bluetooth/data/mapper/MessageActionMapper.kt

```diff
@@ -0,0 +1,48 @@
+package edu.stanford.bdh.engagehf.bluetooth.data.mapper
+
+import edu.stanford.bdh.engagehf.messages.MessagesAction
+import edu.stanford.bdh.engagehf.messages.Questionnaire
+import edu.stanford.bdh.engagehf.messages.VideoSectionVideo
+import javax.inject.Inject
+
+class MessageActionMapper @Inject constructor() {
+
+    companion object {
+        private val videoSectionRegex = Regex("/videoSections/(.+)/videos/(.+)")
+        private val questionnaireRegex = Regex("/questionnaires/(.+)")
+    }
```

Comment:
As proposed in the other comment, let's update both regexes to handle missing initial slash:

```kotlin
private val videoSectionRegex = Regex("/?videoSections/(.+)/videos/(.+)")
private val questionnaireRegex = Regex("/?questionnaires/(.+)")
```